### ISSUE TITLE
Use Hash#fetch in config_file.rb

### ISSUE THREAD

**Issue by duckinator** [rubygems/rubygems#1813]
My current problem is config_file.rb uses if/else instead of Hash#fetch.

https://github.com/rubygems/rubygems/blob/master/lib/rubygems/config_file.rb#L422-L446

The instances of:

```ruby
x = if @hash.key?(:foo)
      @hash[:foo]
    else
      DEFAULT_FOO
    end
```

could be simplified to:

```ruby
x = @hash.fetch(:foo, DEFAULT_FOO)
```

This issue is related to:

  - [ ] Network problems
  - [ ] Installing a library
  - [ ] Publishing a library
  - [ ] The command line `gem`
  - [x] Other

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by duckinator** [comment]
I'll handle this in a bit. Mostly opened this in case I forget.

**Comment by djberg96** [comment]
I don't remember if 1.8.x allows that format. I thought it required a block for a default value. Can you confirm?

**Comment by segiddins** [comment]
Closed by #1824.
